### PR TITLE
[doc] Teach 0-args form of super in Programming FAQ

### PR DESCRIPTION
**Pull request by andresdelfino**
Teach simpler and more maintainable use of super.

### CODE REVIEW

**Comment by rhettinger** [comment]
This change looks good.

While you're at it, the rest of the entry can be fixed up.  In the title, replace "override" with "extend".  Also the text paragraph that follows needs to be revised.

**Comment by andresdelfino** [comment]
@rhettinger had doubts about removing the mention to classic classes, you think that's alright?

**Comment by andresdelfino** [comment]
If you are talking about the next question, this PR deals with it: https://github.com/python/cpython/pull/22177

**Comment by rhettinger** [comment]
I would remove the entire paragraph that starts with "For version prior to 3.0, you may be using classic classes".  That is no longer relevant.

Perhaps replace the paragraph with an explanation of what the zero-arg form of *super()* is actually doing.  It figures out the *self* argument which it uses to derive the *MRO*.  Then it figures out that the current class is *Derived* and uses that information to determine where in the MRO to start seaching.

**Comment by andresdelfino** [comment]
@rhettinger I have made the suggested changes

File: Doc/faq/programming.rst [hunk 1]
```diff
@@ -1518,20 +1518,19 @@ Most :meth:`__setattr__` implementations must modify ``self.__dict__`` to store
...
+           super().meth()  # calls Base.meth
 
-For version prior to 3.0, you may be using classic classes: For a class
-definition such as ``class Derived(Base): ...`` you can call method ``meth()``
-defined in ``Base`` (or one of ``Base``'s base classes) as ``Base.meth(self,
-arguments...)``.  Here, ``Base.meth`` is an unbound method, so you need to
-provide the ``self`` argument.
+In the example, :func:`super` will automatically determine the instance from
+which it was called (the ``self`` value), look up the :term:`method resolution
+order` (MRO) of the instance type (``Derived``), and return the next in line
```
Comment (rhettinger) [inline Doc/faq/programming.rst]: Replace:

    look up the (MRO) of the instance type (Derived)

With:

    look up the (MRO) with ``type(self).__mro__``

Comment (andresdelfino) [reply Doc/faq/programming.rst]: Done!

**Review by rhettinger — APPROVED** [review]

**Pull request merged**